### ISSUE TITLE
Property is possibly undefined even after element access/indexed access

### ISSUE THREAD

**Issue by DaFri-Nochiterov** [Microsoft/TypeScript#16594]
**TypeScript Version:**  2.3.4

**Code**
```ts
interface IGuildCustomize {
    /**
     * Welcome message
     */
    welcome_msg?:string;
    /**
     * Channel for welcome message
     */
    welcome_msg_channel:string;
}
```

```ts
let cz = JSON.parse(guildRow.customize) as IGuildCustomize;
if (cz["welcome_msg"] && cz["welcome_msg_channel"]) {
   let channel = msg.guild.channels.get(cz["welcome_msg_channel"]);
   if (!channel || channel.type !== "text") {
    return;
   }
   (channel as TextChannel).send(cz["welcome_msg"].replace("{usermention}", `<@${msg.author.id}>`).replace("{username}", escapeDiscordMarkdown(msg.author.username)));
}
```

**Expected behavior:** No errors

**Actual behavior:** Errors that value possibly can be `undefined` after `if` check
![image](https://user-images.githubusercontent.com/10401817/27248699-9618ec60-5331-11e7-9934-f9c2fd43c429.png)

**Comment by DaFri-Nochiterov** [comment]
Works fine this way:
![image](https://user-images.githubusercontent.com/10401817/27248731-1e0bf04a-5332-11e7-8a2e-5159dcabb9a3.png)

**Comment by DanielRosenwasser** [comment]
Looks like a duplicate of #10530.

**Comment by mhegazy** [comment]
Automatically closing this issue for housekeeping purposes. The issue labels indicate that it is unactionable at the moment or has already been addressed.
